components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java

```diff
@@ -3578,7 +3578,8 @@ public static void setResourcePermissions(String username, String visibility, St
                     publisherAccessRoles = new StringBuilder(APIConstants.NULL_USER_ROLE_LIST);
                 }
 
-                if (visibility.equalsIgnoreCase(APIConstants.API_GLOBAL_VISIBILITY)) {
+                if (visibility.equalsIgnoreCase(APIConstants.API_GLOBAL_VISIBILITY)
+                        || visibility.equalsIgnoreCase(APIConstants.API_PRIVATE_VISIBILITY)) {
```

Comment:
Its better to use the equalsIgnoreCase, with the constant which is guaranteed not to be null

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/8152